docs/source/working_with_aiida/caching.rst

```diff
@@ -65,88 +115,76 @@ This means that caching is enabled for ``TemplatereplacerCalculation`` and ``Str
+it is also possible to manually *prevent* a specific node from being reused:
 
-1. If you suspect that a node has been cached in error, check that it has a ``_aiida_cached_from`` extra. If that's not the case, it is not a problem of caching.
-2. Get all nodes which match your node, and clear their hash:
+1. For one of the cloned nodes, check that :meth:`~aiida.orm.nodes.Node.get_cache_source` returns a UUID.
+   If it returns `None`, the node was not cloned.
+2. Clear the hashes of all nodes that are considered identical to this node:
 
     .. code:: python
 
         for n in node.get_all_same_nodes():
             n.clear_hash()
-3. Run your calculation again. Now it should not use caching.
-
-If you instead think that there is a bug in the AiiDA implementation, please open an issue (with enough information to be able to reproduce the error, otherwise it is hard for us to help you) in the AiiDA GitHub repository: https://github.com/aiidateam/aiida-core/issues/new.
-
-.. _caching_provenance:
+3. Run your calculation again. The node in question should no longer be reused.
 
-Caching and the Provenance Graph
---------------------------------
 
-The goal of the caching mechanism is to speed up AiiDA calculations by re-using duplicate calculations. However, the resulting provenance graph should be exactly the same as if caching was disabled. This has important consequences on the kind of caching operations that are possible.
-
-The provenance graph consists of nodes describing data, calculations and workchains, and links describing the relationship between these nodes. We have seen that the hash of a node is used to determine whether two nodes are equivalent. To successfully use a cached node however, we also need to know how the new node should be linked to its parents and children.
-
-In the case of a plain data node, this is simple: Copying a data node from an equivalent node should not change its links, so we just need to preserve the links which this new node already has.
-
-For calculations, the situation is a bit more complex: The node can have inputs and creates new data nodes as outputs. Again, the new node needs to keep its existing links. For the outputs, the calculation needs to create a copy of each node and link these as its outputs. This makes it look as if the calculation had produced these outputs itself, without caching.
-
-Finally, workchains can create links not only to nodes which they create themselves, but also to nodes created by a calculation that they called, or even their ancestors. This is where caching becomes impossible. Consider the following example (using workfunctions for simplicity):
-
-.. code:: python
+.. _caching_limitations:
 
-    from aiida.engine import workfunction
-    from aiida.orm import Int
+Limitations
+-----------
 
-    @workfunction
-    def select(a, b):
-        return b
+#. While caching saves unnecessary computations, the current implementation does not yet save disk space:
```

Comment:
This is not so much a limitation (as is the uncachability of workflow processes) as an implementation inefficiency.
I would maybe rather add this as a note or warning in an earlier section where we introduce the caching and its benefits. In any case I would not add it as the first item here, but rather start with the uncachability of workflows

Reply:
I've now moved it to the end of the list.
I still consider it a limitation, since it is something that users might naturally expect to happen when they read "caching" (I would).

Reply:
I am not sure I agree. This relates to how we define "caching" in AiiDA and boils down to the same discussion as in the other thread about the configuration file enabling the caching of `Data` nodes. Since caching in AiiDA does explicitly _not_ mean the deduplication of content storage, but just and only just the skipping of executing calculations that have already been executed before, the absence of content deduplication is **not** a limitation. It is just an efficiency of the database implementation. Note that this problem exists completely independent of AiiDA. If you store a `Data` node with the same content a million times, its content will be stored on disk a million times. This is why I am a strong proponent to making more clear what we understand with "caching" in AiiDA.

Reply:
Sure, we both agree that the lack of content deduplication is not a limitation of caching as we define it here.
However, I think most people will not have such a clear separation of these two concepts in their mind, and they might expect deduplication to happen as well.

It was already the last item in the list - I've now made it a comment after the list (no longer a list item). I think it should be kept.

Reply:
I agree that it is important and should be kept, I was just suggesting to move it higher up in the introduction. But like this is fine as well, thanks